### ISSUE TITLE
Operator only accepts strings in `helmValues`

### ISSUE THREAD

**Issue by nbkrause** [datawire/ambassador-operator#4]
The operator only accepts strings in the `helmValues` field of an `AmbassadorInstallation`.

This does not work since a large number of the configurable options in the Chart `values.yaml` is a more complex YAML object. See the table here: https://github.com/datawire/ambassador-chart#configuration

We need to make this more flexible since people should be able to configure anything that is configurable in the helm chart

**Comment by eserozvataf** [comment]
same here:

it says:
```
Error: AmbassadorInstallation.getambassador.io "ambassador" is invalid: [spec.helmValues.crds: Invalid value: "object": spec.helmValues.crds in body must be of type string: "object", spec.helmValues.namespace:
 Invalid value: "object": spec.helmValues.namespace in body must be of type string: "object", spec.helmValues.rateLimit: Invalid value: "object": spec.helmValues.rateLimit in body must be of type string: "obje
ct", spec.helmValues.rbac: Invalid value: "object": spec.helmValues.rbac in body must be of type string: "object", spec.helmValues.redis: Invalid value: "object": spec.helmValues.redis in body must be of type
string: "object", spec.helmValues.replicaCount: Invalid value: "integer": spec.helmValues.replicaCount in body must be of type string: "integer", spec.helmValues.adminService: Invalid value: "object": spec.hel
mValues.adminService in body must be of type string: "object", spec.helmValues.authService: Invalid value: "object": spec.helmValues.authService in body must be of type string: "object", spec.helmValues.autosc
aling: Invalid value: "object": spec.helmValues.autoscaling in body must be of type string: "object"]
```

**Comment by inercia** [comment]
Serialization for complex data structures is cyurrently a bit more complicated. We cannot pass a YAML subtree, so we must build the tree by repeting the elements, like this:

```yaml
apiVersion: getambassador.io/v2
kind: AmbassadorInstallation
metadata:
  name: ambassador
spec:
  version: "1.*"
  updateWindow: "* 0-6 * * * SUN"
  helmValues:
    service.ports[0].name: "http"
    service.ports[0].port: "80"
    service.ports[0].targetPort: "8080"
    service.ports[1].name: "https"
    service.ports[1].port: "443"
    service.ports[1].targetPort: "8443"
```

We will investigate some way users can provided their own `values.yaml` files...

**Comment by inercia** [comment]
Closed in favor of #6
